src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1393,110 +1339,217 @@ internal static unsafe bool TryParseUInt64(ReadOnlySpan<char> s, NumberStyles st
             if (IsWhite(num))
             {
                 if ((styles & NumberStyles.AllowTrailingWhite) == 0) goto FalseExit;
-                for (index++; index < source.Length; index++)
+                for (index++; index < value.Length; index++)
                 {
-                    if (!IsWhite(source[index])) break;
+                    if (!IsWhite(value[index])) break;
                 }
-                if ((uint)index >= (uint)source.Length) goto DoneAtEndButPotentialOverflow;
+                if ((uint)index >= (uint)value.Length) goto DoneAtEndButPotentialOverflow;
             }
 
-            if (!TrailingZeros(source, index)) goto FalseExit;
+            if (!TrailingZeros(value, index)) goto FalseExit;
             
             goto DoneAtEndButPotentialOverflow;
         }
 
-        internal static unsafe decimal ParseDecimal(ReadOnlySpan<char> value, NumberStyles options, NumberFormatInfo numfmt)
+        /// <summary>Parses ulong limited to styles that make up NumberStyles.HexNumber.</summary>
+        private static bool TryParseUInt64HexNumberStyle(
+            ReadOnlySpan<char> value, NumberStyles styles, NumberFormatInfo info, out ulong result, ref bool failureIsOverflow)
+        {
+            Debug.Assert((styles & ~NumberStyles.HexNumber) == 0, "Only handles subsets of HexNumber format");
+            Debug.Assert(!failureIsOverflow, $"failureIsOverflow should have been initialized to false");
+
+            if ((uint)value.Length < 1) goto FalseExit;
+
+            bool overflow = false;
+            int index = 0;
+            int num = value[0];
+            int numValue = 0;
+
+            // Skip past any whitespace at the beginning.  
+            if ((styles & NumberStyles.AllowLeadingWhite) != 0 && IsWhite(num))
+            {
+                index++;
+                while (true)
+                {
+                    if ((uint)index >= (uint)value.Length) goto FalseExit;
+                    num = value[index];
+                    if (!IsWhite(num)) break;
+                    index++;
+                }
+            }
+
+            long answer = 0;
+            int[] charToHexLookup = s_charToHexLookup;
+
+            if ((uint)num < (uint)charToHexLookup.Length && charToHexLookup[num] != 0xFF)
+            {
+                // Skip past leading zeros.
+                if (num == '0')
+                {
+                    do
+                    {
+                        index++;
+                        if ((uint)index >= (uint)value.Length) goto DoneAtEnd;
+                        num = value[index];
+                    } while (num == '0');
+                    if ((uint)num >= (uint)charToHexLookup.Length || charToHexLookup[num] == 0xFF) goto HasTrailingChars;
+                }
+ 
+                // Parse up through 16 digits, as no overflow is possible
```

Comment:
"Parse up **through** 16 digits..." => "Parse up **to** 16 digits..."?

Reply:
"Through" is correct.

Reply:
ok.